### ISSUE TITLE
Decompressing super voxel meshes in parallel has problems on Linux

### ISSUE THREAD

**Issue by hubbardp** [janelia-flyem/NeuTu#100]
When Neu3 loads a body's super voxels, it has to do the Draco decompression on for each super voxel mesh.  To improve performance, it decompresses the meshes in parallel.  It is working well on OS X, but it turns out that the implementation can fail on Linux.

The problem seems to be in the OS-specific implementation of the C++11 std::async function.  It is supposed to relieve the programmer of the need to think about explicit threads, but that would be true only if it really used a thread pool.  That seems to be the case on OS X, but on Linux, it seems to create new threads for each call.  Eventually, it fails when it reaches the maximum number of threads supported by the system.

**Comment by stuarteberg** [comment]
For posterity, here is a program that demonstrates the difference in behavior of `std::async` between Mac and Linux (i.e. the difference between GCC's `libstdc++` and Clang/LLVM's `libc++`):

The key findings are:

1. With clang/`libc++`, the default `launch` policy is to use `async` (multi-thread), whereas with GCC/`libstdc++`, the defalut is `deferred` (single thread).

    ```
    # Linux, default launch
    
    $ ./test_async.out
    Tasks were executed on 1 unique threads.
    ```

2. If you explicitly set an `async` launch policy, the thread pool behavior is different on the two platforms.  With clang/`libc++`, a pool of threads is created and re-used:

    ```
    # Mac, async launch
    
    $ ./test_async.out
    Tasks were executed on 90 unique threads.
    ```

    ...but with gcc/`libstdc++` a new thread is created for EVERY task.  That could easily overrun the system if you have thousands of tasks:

    ```
    # Linux, async launch
    
    $ ./test_async.out
    Tasks were executed on 1000 unique threads.
    ```

The demonstration program is given below.

<details>
<summary>Expand to see source.</summary>

```c++
#include <iostream>
#include <future>
#include <thread>
#include <chrono>
#include <mutex>
#include <vector>
#include <unordered_set>

// To compile and run on Mac:
// $ clang++ -std=c++14 -o test_async.out test_async.cpp
// $ ./test_async.out
//
// On Linux:
// $ g++ -pthread -std=c++14 -o test_async.out test_async.cpp
// $ ./test_async.out

//
// Parameters for this test.
//
int NUM_TASKS = 1000;
int DELAY_MS = 1;
bool EXPLICIT_LAUNCH_POLICY = false;

std::mutex mut;
std::unordered_set<std::thread::id> thread_ids;

// Pause for a bit, then record the current thread ID in the global set.
int append_thread_id()
{
	std::chrono::milliseconds delay(DELAY_MS);
	std::this_thread::sleep_for(delay);

	auto id = std::this_thread::get_id();
	std::lock_guard<std::mutex> guard(mut);
	thread_ids.insert(id);
	return 0;
};


int main()
{
	std::vector< std::future<int> > tasks;

	for ( int i = 0; i < NUM_TASKS; ++i )
	{
		if (EXPLICIT_LAUNCH_POLICY)
		{
			auto task = std::async(std::launch::async, &append_thread_id);
			tasks.push_back(std::move(task));
		}
		else
		{
			// Use platform-defined default launch policy
			// Apparently Mac and Linux use different defaults!
			auto task = std::async(&append_thread_id);
			tasks.push_back(std::move(task));
		}

	}

	for ( auto & task : tasks )
	{
		task.wait();
	}

	std::cout << "Tasks were executed on " << thread_ids.size() << " unique threads." << std::endl;

	return 0;
}
```
</details>

**Comment by tingzhao** [comment]
Another option is to try Qt's concurrent functions. But I'm not sure if it will have the same problem.

**Comment by hubbardp** [comment]
Fixed by changing from std::async to QtConcurrent::mapped, in the branch "neu3-read-mesh-archive-qtconcurrent":
https://github.com/janelia-flyem/NeuTu/tree/neu3-read-mesh-archive-qtconcurrent

**Comment by hubbardp** [comment]
Now in the "testCleaving" release for Linux:
/groups/flyem/data/shared/testCleaving/bin/neu3

**Comment by hubbardp** [comment]
Merge of the "neu3-body-cleave-warning-suppress" branch (based on the "neu3-read-mesh-archive-qtconcurrent" branch)  into the "develop" branch is pending:
https://github.com/janelia-flyem/NeuTu/pull/104

**Comment by hubbardp** [comment]
Now in the "develop" branch.
